gammapy/modeling/models/temporal.py

```diff
@@ -454,97 +445,145 @@ class LightCurveTemplateTemporalModel(TemporalModel):
 
     tag = ["LightCurveTemplateTemporalModel", "template"]
 
-    def __init__(self, table, filename=None):
-        self.table = table
-        if filename is not None:
-            filename = str(make_path(filename))
-        self.filename = filename
+    _t_ref_default = Time("2000-01-01")
+    t_ref = Parameter("t_ref", _t_ref_default.mjd, unit="day", frozen=False)
+
+    def __init__(self, map, t_ref=None):
+
+        if (map.data < 0).any():
+            log.warning("Map has negative values. Check and fix this!")
+
+        self.map = map.copy()
+        if t_ref:
+            self.t_ref.value = Time(t_ref, format="MJD").mjd
+        if "ENERGY" in map.geom.axes.names:
+            self.is_energy_dependent = True
```

Comment:
Please tun this into a property, such that it cannot be modified by users. Please also use https://docs.gammapy.org/0.20.1/api/gammapy.maps.Geom.html#gammapy.maps.Geom.has_energy_axis, to avoid hard-coding axis names.